src/shogun/base/SGObject.h

```diff
@@ -1011,5 +1024,69 @@ class CSGObject
 	/** Subscriber used to call onNext, onComplete etc.*/
 	SGSubscriber* m_subscriber_params;
 };
+
+#ifndef SWIG
+
+namespace sgobject_get_details
```

Comment:
other header file? sgobject_utils.h or anything that is not making this file bigger

Reply:
actually realised this might be tricky as I am calling sgobject methods so need the type, but also need to include the header files.